src/App.js

```diff
@@ -55,6 +57,92 @@ export default function App() {
     }
   }
 
+  const checkIfCorrectBlockchain = async () => {
+    try {
+      const { ethereum } = window;
+
+      if (!ethereum) {
+        alert("Get Metamask!");
+        return;
+      }
+      const provider = new ethers.providers.Web3Provider(ethereum);
+      const networkInfo = await provider.getNetwork();
+      const { chainId } = networkInfo;
+      if (!(chainId === 43113 || chainId === 43114)) {
+        setShowDialogWrongBlockchain(true);
+      }
+    } catch (error) {
+      
+    }
+  }
+
+  const updateOnBlockchainChanged = useCallback(async () => {
+    try {
+      const { ethereum } = window;
+
+      if (!ethereum) {
+        alert("Get Metamask!");
+        return;
+      }
+
+      ethereum.on('chainChanged', (chainId) => {
+        console.log(chainId);
+        // Handle the new chain.
+        if (!(chainId === 43113 || chainId === 43114)) {
```

Comment:
We shouldn't hardcode any data (here the id of the fuji and avalanche chains).
Instead, we should create a `constants` directory under `src` and a file `address.js` where we store these values :
```js
export const FUJI_TESTNET_ID = 43113;
export const AVALANCHE_MAINNET_ID = 43114;
```